tests/storage/guestfs.go

```diff
@@ -80,36 +77,31 @@ var _ = SIGDescribe("[rfe_id:6364][[Serial]Guestfs", func() {
 			Expect(guestfsCmd.Execute()).ToNot(HaveOccurred())
 		}()
 		// Waiting until the libguestfs pod is running
-		Eventually(func() corev1.PodPhase {
+		Eventually(func() bool {
 			pod, _ := virtClient.CoreV1().Pods(util.NamespaceTestDefault).Get(context.Background(), podName, metav1.GetOptions{})
-			return pod.Status.Phase
-		}, 90*time.Second, 2*time.Second).Should(Equal(k8sv1.PodRunning))
-		// Verify that the appliance has been extracted before running any tests
-		// We check that all files are present and tar is not running
+			ready := false
+			for _, status := range pod.Status.ContainerStatuses {
+				if status.State.Running != nil {
+					return true
+				}
+			}
+			return ready
+
+		}, 90*time.Second, 2*time.Second).Should(BeTrue())
+		// Verify that the appliance has been extracted before running any tests by checking the done file
 		Eventually(func() bool {
-			output, _, err := execCommandLibguestfsPod(podName, []string{"ls", "-m", "/usr/local/lib/guestfs/appliance"})
-			Expect(err).ToNot(HaveOccurred())
-			if !strings.Contains(output, APPLIANCE_FILES) {
+			output, _, err := execCommandLibguestfsPod(podName, []string{"ls", "/usr/local/lib/guestfs/done"})
+			if err != nil {
 				return false
 			}
-			output, _, err = execCommandLibguestfsPod(podName, []string{"ps", "-e", "-o", "comm="})
-			Expect(err).ToNot(HaveOccurred())
-			if strings.Contains(output, "tar") {
-				return false
+			if strings.Contains(output, "/usr/local/lib/guestfs/done") {
```

Comment:
I think this check is redundant. `ls` fails if the targeted file does not exist and `err` should then non-nil, right?